### ISSUE TITLE
ActivitySpec and SensorSpec endpoints require implementation

### ISSUE THREAD

**Issue by SuruPa00** [BIDMCDigitalPsychiatry/LAMP-platform#361]
See the two files here related to ActivitySpec: https://github.com/BIDMCDigitalPsychiatry/LAMP-server/blob/55deb584c4f13cd6856356a5884166eed293b4eb/src/repository/mongo/ActivitySpecRepository.ts#L18

And SensorSpec: https://github.com/BIDMCDigitalPsychiatry/LAMP-server/blob/55deb584c4f13cd6856356a5884166eed293b4eb/src/repository/mongo/SensorSpecRepository.ts#L18

Both return errors that say unimplemented, can you look at these and correctly implement the endpoints?

**Comment by avaidyam** [comment]
@Linoy339 Could you add the implementation detail for both CouchDB and MongoDB? It should be quite simple in both cases.

**Comment by Linoy339** [comment]
@avaidyam @SuruPa00 . We can update it ASAP and let you know. 
Thanks

**Comment by Linoy339** [comment]
@avaidyam . Is the both functions to be implemented for bulk insert or single one?

**Comment by avaidyam** [comment]
All six functions across two files should be implemented.

**Comment by Linoy339** [comment]
Okay @avaidyam

**Comment by ryan-hays** [comment]
I see @avaidyam merged this. Looks good! Can we create a new release so that the endpoints are available in the production server?

**Comment by ryan-hays** [comment]
Also, I noticed in testing that `LAMP.ActivitySpec.create` has scope across the server, even if one is logged in with `Researcher` credentials; `LAMP.ActivitySpec.delete` is only available to admin.

Is this intended behavior? @avaidyam Would it be better to selectively allow ActivitySpec creation within the scope of individual researchers?

**Comment by avaidyam** [comment]
@ryan-hays Good catch! @Linoy339 Create/Update/Delete should ONLY be available to administrator level. Could you update this?

**Comment by Linoy339** [comment]
Sure @avaidyam .Let me check on it

**Comment by Linoy339** [comment]
@avaidyam @ryan-hays . Create api is made limited to administrator. 
Update/Delete operation was already limited to administrator
